azure/bucket.ts

```diff
@@ -13,29 +13,124 @@
 // limitations under the License.
 
 import * as azure from "@pulumi/azure";
+import * as serverless from "@pulumi/azure-serverless";
 import * as cloud from "@pulumi/cloud";
 import * as pulumi from "@pulumi/pulumi";
+import * as stream from "stream";
+import * as shared from "./shared";
 
 export class Bucket extends pulumi.ComponentResource implements cloud.Bucket {
+    public readonly container: azure.storage.Container;
+
+    // deployment-time api:
+    public onPut: (name: string, handler: cloud.BucketHandler, filter: cloud.BucketFilter | undefined) => void;
+    public onDelete: (name: string, handler: cloud.BucketHandler, filter: cloud.BucketFilter | undefined) => void;
+
+    // run-time api
     public get: (key: string) => Promise<Buffer>;
     public put: (key: string, contents: Buffer) => Promise<void>;
     public delete: (key: string) => Promise<void>;
 
     public constructor(name: string, opts?: pulumi.ResourceOptions) {
         super("cloud:bucket:Bucket", name, {}, opts);
 
-        this.get = _ => { throw new Error("Method not implemented."); };
-        this.put = _ => { throw new Error("Method not implemented."); };
-        this.delete = _ => { throw new Error("Method not implemented."); };
+        const preventDestroy = opts && opts.protect;
+
+        const resourceGroupName = shared.resourceGroupName;
+        const storageAccount = shared.getGlobalStorageAccount();
```

Comment:
Current design decisions: 
1. there is a global resource group that can also be specified by config.
2. there is a global cloud infrastructure/storage account.  Currently there is no way to configure this.

Reply:
> We could consider making it so if you don't provide one we create one for you.

Yes - I think we should do this.  I expect the normal case here will be the user wants this to be it's own resource group.

Reply:
ok.  let me try that.